extract-docker-tag/main.go

```diff
@@ -71,20 +72,55 @@ func extract(action *githubactions.Action) (result result, err error) {
 		branch := action.Getenv("GITHUB_HEAD_REF")
 		parts = strings.Split(strings.ToLower(branch), "/") // for branches like "dependabot/submodules/XXX"
 		result.tag = "pr-" + parts[len(parts)-1]
+
 	case "push", "schedule", "workflow_run":
-		branch := action.Getenv("GITHUB_REF_NAME")
-		if branch == "main" { // build on pull_request/pull_request_target for other branches
+		refName := action.Getenv("GITHUB_REF_NAME")
+		refType := action.Getenv("GITHUB_REF_TYPE")
+
+		result.tag, err = getTag(refName, refType)
+		if err != nil {
+			return
+		}
+
+		// build on pull_request/pull_request_target for other branches
+		switch refType {
+		case "branch":
+			result.name += "-dev"
+		case "tag":
 			result.name += "-dev"
-			result.tag = strings.ToLower(branch)
+		default:
+			panic("unreachable code")
 		}
+	default:
+		panic("unreachable code")
 	}
+
 	if result.tag == "" {
 		err = fmt.Errorf("failed to extract tag for event %q", event)
 		return
 	}
-
-	// set ghcr
 	result.ghcr = fmt.Sprintf("ghcr.io/%s/%s:%s", result.owner, result.name, result.tag)
 
 	return
 }
+
+func getTag(refName, refType string) (tag string, err error) {
```

Comment:
It seems that function code would be easier to read without named parameters.

Reply:
it saves in return rows readability  -_- WDYT?

Reply:
https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters
Named results is a smell of something tricky going on: panic handling with `recover()`, setting values in `defer`s, etc. If we don't have anything like that, let's drop it.

(It might also be true for the `extract` function too. I don't remember why it had named results.)

Reply:
@AlekSi @w84thesun named parameters removed